src/main/java/Duke.java

```diff
@@ -1,10 +1,194 @@
+import java.util.Scanner;
+
+
+/**
+ * Main class that runs the chat bot. Tasks are managed by the TaskManager
+ * Duke provides the extraction of action words, descriptions and due dates
+ * Duke also provides methods to print useful messages
+ */
 public class Duke {
+
+    public static final String EXIT_COMMAND = "BYE";
+    public static final String LIST_COMMAND = "LIST";
+    public static final String COMPLETE_COMMAND = "DONE";
+    public static final String TODO_COMMAND = "TODO";
+    public static final String DEADLINE_COMMAND = "DEADLINE";
+    public static final String EVENT_COMMAND = "EVENT";
+    public static final String LOGO = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+    public static final String DEADLINE_PREFIX = "/by";
+    public static final String EVENT_PREFIX = "/at";
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        // Startup and Initializations
+        Scanner sc = new Scanner(System.in);
+        String currentUserInput = "";
+        TaskManager taskManager = new TaskManager();
+        String actionWord;
+        String description;
+        String timing;
+        int timingIndex;
+        int descriptionIndex;
+
+        // Introduce the bot after startup
+        printIntroduction(LOGO);
+        while (!currentUserInput.equalsIgnoreCase(EXIT_COMMAND)) {
+            currentUserInput = sc.nextLine();
+            // actionWord will always be the first word in the sentence
+            actionWord = currentUserInput.toUpperCase().split(" ")[0];
+            // Find the first " ". If it does not exist, the command would not use the index anyway. If it does
+            // + 1 to remove the space and get the first index of the description.
+            descriptionIndex = currentUserInput.indexOf(" ") + 1;
+
+            String[] outputMessages;
+            switch (actionWord) {
+            case EXIT_COMMAND:
+                break;
+
```

Comment:
I am not entirely sure whether an empty line counts as a violation, but it might be safer to not leave the line breaks?